makefile

```
@@ -1,6 +1,8 @@
 
 BOUT_TOP	= ../..
 
+BOUT_GIT_COMMIT:=$(shell git --git-dir=$(BOUT_TOP)/.git rev-parse --short HEAD >> BOUT_commit)
```

It looks like this isn't used anywhere; can you see how STORM gets the commit into the executable?

---

In STORM we have
```
CXXFLAGS += -DSTORM_VERSION=\"$(GIT_VERSION)\"
```
in the `makefile`.
Although this looks like it's getting the commit for BOUT rather than HERMES, but BOUT saves it's commit to the log and dump files already.

I'd suggest
```suggestion
GIT_VERSION := $(shell git describe --abbrev=40 --dirty --always --tags)
CXXFLAGS += -DHERMES_VERSION=\"$(GIT_VERSION)\"
```
or similar (depending how you like your git hash to be formatted).